fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -676,7 +915,8 @@ private boolean isJoinMatch(OptExpression queryExpression,
             // Use traverse order to check whether all joins' order and operator are exactly matched.
             List<JoinOperator> queryJoinOperators = MvUtils.getAllJoinOperators(queryExpression);
             List<JoinOperator> mvJoinOperators = MvUtils.getAllJoinOperators(mvExpression);
-            return queryTables.equals(mvTables) && queryJoinOperators.equals(mvJoinOperators);
+            return (queryTables.equals(mvTables) && queryJoinOperators.equals(mvJoinOperators))
+                    || computeCompatibility(queryExpression, mvExpression);
```

Comment:
use `computeCompatibility` directly? no need to keep original codes?
```
return  computeCompatibility(queryExpression, mvExpression);
```

Reply:
done